.github/workflows/deployment.yml

```diff
@@ -0,0 +1,103 @@
+name: Tests and publishing
+env:
+  FORCE_COLOR: 1
+on:
+  push:
+    branches:
+      - master
+      - main
+      - develop
+      - stage
+  pull_request:
+    branches:
+      - master
+      - main
+      - stage
+jobs:
+  test_linux:
+    name: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+        # os: [ubuntu-18.04, ubuntu-20.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - uses: microsoft/playwright-github-action@v1
+      - uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install dependencies
+        run: npm ci
+      - name: Run tests
+        run: npm test
+  test_win:
+    name: "Windows"
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - uses: microsoft/playwright-github-action@v1
+      - uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install dependencies
+        run: npm ci
+      - name: Run tests
+        run: npm test
+  tag:
+    name: "Publishing release"
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master'
```

Comment:
Would it be possible to release a `patch` of a previous `major.minor`?

Reply:
Yes, it depends on the version that's set in the `package.json` file

Reply:
QQ when we release a `patch` of a previous `major.minor` we create a branch from the `tag` of that release, right? For example, imagine we need to do a fix on `6.4.12` version. We create a branch `build/v6.4.13` (or something like that) from that tag `6.4.12`. How could I use this GitHub action to do a release if we only release from main/master branch?

Reply:
That's a good point. I'll add the support branches